src/platform/cmx/chal/cos_config.h

```diff
@@ -0,0 +1,110 @@
+
+#define COS_MEM_KERN_VA_SZ COS_MEM_KERN_PA_SZ     /* just the same as the physical address */
+
+/* To get more memory, we need many PTE caps in the captbl. So give
+ * multiple pages to it. 5 is enough for 512 MBs.*/
+#define BOOT_CAPTBL_NPAGES 1
+
+#define BOOT_COMP_MAX_SZ   (1<<14) /* 16 kB for the booter component */
+#define NUM_CPU                1
+#define CPU_TIMER_FREQ         100 // set in your linux .config
+#define RUNTIME                3 // seconds
+
+/* The kernel quiescence period = WCET in Kernel + WCET of a CAS. */
+#define KERN_QUIESCENCE_PERIOD_US 500
+#define KERN_QUIESCENCE_CYCLES (KERN_QUIESCENCE_PERIOD_US * 4000)
+#define TLB_QUIESCENCE_CYCLES  (4000 * 1000 * (1000 / CPU_TIMER_FREQ))
+
+// After how many seconds should schedulers print out their information?
+#define SCHED_PRINTOUT_PERIOD  100000
+#define COMPONENT_ASSERTIONS   1 // activate assertions in components?
+
+//#define FPU_ENABLED
+#define FPU_SUPPORT_FXSR       1   /* >0 : CPU supports FXSR. */
+
+/* the CPU that does initialization for Composite */
+#define INIT_CORE              0
+#define NUM_CPU_COS            (NUM_CPU > 1 ? NUM_CPU - 1 : 1)
+
+/* Composite user memory uses physical memory above this. */
+#define COS_MEM_START          COS_MEM_KERN_PA
+
+/* NUM_CPU_SOCKETS defined in cpu_ghz.h. The information is used for
+ * intelligent IPI distribution. */
+#define NUM_CORE_PER_SOCKET    (NUM_CPU / NUM_CPU_SOCKETS)
+
+// cos kernel settings
+#define COS_PRINT_MEASUREMENTS 1
+#define COS_PRINT_SCHED_EVENTS 1
+#define COS_ASSERTIONS_ACTIVE  1
+
+/*** Console and output options ***/
+/*
+ * Notes: If you are using composite as high priority and no idle to
+ * linux, then the shell output will not appear until the Composite
+ * system has exited.  Thus, you will want to make the memory size
+ * large enough to buffer _all_ output.  Note that currently
+ * COS_PRINT_MEM_SZ should not exceed around (1024*1024*3).
+ *
+ * If you have COS_PRINT_SHELL, you _will not see output_ unless you
+ * run
+ * $~/transfer/print
+ * after
+ * # make
+ * but before the runscript.
+ */
+/* print out to the shell? */
+#define COS_PRINT_SHELL   1
+/* how much should we buffer before sending an event to the shell? */
+#define COS_PRINT_BUF_SZ  128
+/* how large should the shared memory region be that will buffer print data? */
+#define COS_PRINT_MEM_SZ  (4096)
+
+/* print out to dmesg? */
+/* #define COS_PRINT_DMESG 1 */
+
+/**
+ * Configuration to enable/disable functionality in Kernel.
+ */
+/*
+#define ENABLE_VGA
+#define ENABLE_SERIAL*/
+
+
+#define CFORCEINLINE /* __attribute__((always_inline)) */
+#define COS_SYSCALL /* __attribute__((regparm(0))) __attribute__((section("__ipc_entry"))) */
+/* The whole stack frame's pointer is in r5. Now the parameter passing is correct */
+#define COS_SYSCALL_PASS_PARAM() __asm__ __volatile__("str r0,%[_regs] \n"::[_regs]"m"(regs):"memory")
```

Comment:
This is removed after a revisit of the system handler calling code. Currently we don't need this macro anymore.